### PR TITLE
NSRange and Attributed Strings

### PR DESCRIPTION
**Pull request by paulrolfe**
Add some tests and fixed some tests.

### CODE REVIEW

File: SwiftWisdom/Core/StandardLibrary/NSAttribitedString/NSMutableAttributedString+Format.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    
+    /**
+     Create a mutable attributed string with the given collection of NSAttributedString
+     
+     - parameter string: formatted string where `%@` represents an argument.
+     - parameter args:   comma separated list of arguments (NSAttributedString)
+     
+     - returns: NSMutableAttributedString with arguments inserted according to format.
+     */
+    private convenience init(stringWithFormat string: String, withArguments args: [NSAttributedString]) {
```
Comment (loganwright) [inline SwiftWisdom/Core/StandardLibrary/NSAttribitedString/NSMutableAttributedString+Format.swift]: any reason to have this `private`? Might be useful to be able to call w/ an array as well.  I might change the above `_ args` to match this signature so the overload is clear that they can use either.  

Overloading w/ array seems to be convention until splatting is explicitly supported

File: SwiftWisdomTests/SwiftWisdomTests.swift [hunk 2]
```diff
@@ -28,3 +28,30 @@ class GetConstraintTests: XCTestCase {
...
+            attributes: [NSForegroundColorAttributeName : UIColor.redColor()]
+        )
+        let testString = NSMutableAttributedString(stringWithFormat: "This color is %@", redText)
+        let otherTestString = NSMutableAttributedString(
+            stringWithFormat: "This color is %@",
+            applyingAttributes: [NSForegroundColorAttributeName : UIColor.redColor()],
+            toArgs: "RED"
+        )
+        
+        XCTAssertEqual(testString, otherTestString)
```
Comment (loganwright) [inline SwiftWisdomTests/SwiftWisdomTests.swift:50]: Does this test the substring colors? ie: if we made one `blueColor`? would it fail?

**Comment by loganwright** [comment]
Cool, good additions.

Add `ip_` prefix to all functions and :shipit:

File: SwiftWisdomTests/SwiftWisdomTests.swift (hunk 2 above)
Comment (paulrolfe) [inline SwiftWisdomTests/SwiftWisdomTests.swift:50]: Yup, but I like that test, I added it in.

File: SwiftWisdom/Core/StandardLibrary/NSAttribitedString/NSMutableAttributedString+Format.swift (hunk 1 above)
Comment (paulrolfe) [inline SwiftWisdom/Core/StandardLibrary/NSAttribitedString/NSMutableAttributedString+Format.swift]: I just didn't know overloading was preferred. I'll do that.

**Pull request merged**